src/screens/ForgotPasswordReset/index.tsx

```diff
@@ -0,0 +1,200 @@
+import { ApolloError } from '@apollo/client';
+import { useNavigation } from '@react-navigation/native';
+import {
+  Box,
+  Heading,
+  Center,
+  FormControl,
+  Input,
+  Button,
+    VStack,
+    Text
+} from 'native-base';
+import { useState } from 'react';
+import { useForm, Controller } from 'react-hook-form';
+
+import {
+  ConfirmForgotPasswordInput,
+  useForgotPasswordResetMutation,
+} from '../../generated/graphql';
+
+interface FormProps {
+  email: string;
+  password: string;
+  confirmationCode: string;
+}
+
+function ForgotPasswordReset() {
+    const [errorMessage, setErrorMessage] = useState<string>('');
+    const [successMessage, setSuccessMessage] = useState<string | undefined>('');
+
+    const navigation = useNavigation<any>();
+
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormProps>();
+  const [forgotPasswordResetMutation, { loading, error }] =
+    useForgotPasswordResetMutation({
+      onCompleted(res) {
+        if (res.confirmForgotPassword?.success === false) {
+          setErrorMessage('An error occurred during password reset.');
+        } else {
+          setSuccessMessage('Password reset successful. Rerouting to Login.');
+          setTimeout(() => {
+            navigation.navigate('LoginScreen');
+          }, 2500); 
+        }
+      }
+      });
+ 
+
+  const emailPattern = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
+
+  const onSubmit = async (data: ConfirmForgotPasswordInput) => {
+    const { email, password, confirmationCode } = data;
+
+    await forgotPasswordResetMutation({
+      variables: {
+        input: {
+          email,
+          password,
+          confirmationCode,
+        },
+      },
+    });
+  };
+  const getConfirmationCodeErrorMessage = (): string | undefined => {
+    if (errors?.confirmationCode) {
+      return 'Invalid confirmation code, please try again';
+    }
+    if (error instanceof ApolloError) {
+      return 'Invalid confirmation code, please try again';
+    }
+    return undefined;
+  };
+
+  return (
+    <Center flex={1} py="10">
+      <Box>
+        <Heading
+          size="lg"
+          fontWeight="600"
+          color="coolGray.800"
+          _dark={{
+            color: 'warmGray.50',
+          }}
+        >
+          Welcome to CraftPile!
```

Comment:
Can we change this heading to "Reset Password"

Reply:
Yup. Done.